crates/lsp/src/lsp.rs

```diff
@@ -97,16 +100,72 @@ pub enum Subscription {
 #[derive(Serialize, Deserialize)]
 pub struct Request<'a, T> {
     jsonrpc: &'static str,
-    id: usize,
+    #[serde(deserialize_with = "deserialize_id", serialize_with = "serialize_id")]
+    id: RequestId,
     method: &'a str,
     params: T,
 }
 
+/// ID is either string or number
+
+#[derive(Debug, Clone, Eq, PartialEq, Hash)]
+pub enum RequestId {
+    Int(usize),
+    Str(String),
+}
+
+// Custom deserializer for the `Id` enum
+fn deserialize_id<'de, D>(deserializer: D) -> Result<RequestId, D::Error>
```

Comment:
Could we mark `RequestId` as `#serde[untagged]` instead of writing our own deserialization/serialization functions?

Reply:
Did not know about that, I will try it out and reiterate.